src/Features/Core/Portable/QuickInfo/IndentationHelper.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.QuickInfo
+{
+    // Reproduces logic in IProjectionBufferFactoryServiceExtensions (editor layer) 
```

Comment:
definitely concerning.  is there a reason that we can't just have the logic exist here and have IProjectionBufferFactoryServiceExtensions  call this instead?

Reply:
Unfortunately, this is not trivial to do. `IProjectionBufferFactoryServiceExtensions` operates directly on editor primitives. Converting those to use this is not trivial. However, this is really just there for tests.